### PR TITLE
[DPE-2144] Refactoring: refactor user related code

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## About 

The goal of this PR is to refactor users related code in K8s charm.

## Related issues
DPE-2144, DPE-2071

### CODE REVIEW

File: lib/charms/mongodb/v0/users.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+"""Users configuration for MongoDB."""
```
Comment (dmitry-ratushnyy) [inline lib/charms/mongodb/v0/users.py:1]: The content of this file is taken from VM charm library as-is

File: src/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
```
Comment (dmitry-ratushnyy) [inline src/exceptions.py:1]: The content of this file is taken from VM charm library as-is

File: src/charm.py [hunk 3]
```diff
@@ -101,34 +103,13 @@ def mongodb_config(self) -> MongoDBConfiguration:
...
-            replset=self.app.name,
-            database="admin",
-            username="operator",
-            password=self.get_secret("app", "operator_password"),
-            hosts=set(hosts),
-            roles={"default"},
-            tls_external=external_ca is not None,
-            tls_internal=internal_ca is not None,
-        )
+        return self._get_mongodb_config_for_user(OperatorUser, set(hosts))
```
Comment (Mehdi-Bendriss) [inline src/charm.py:106]: It seems that you always cast to `Set`, consider changing the method's signature with `List` and cast to `Set` from within the said method

File: src/charm.py [hunk 4]
```diff
@@ -101,34 +103,13 @@ def mongodb_config(self) -> MongoDBConfiguration:
...
-            database="",
-            username="monitor",
-            # MongoDB Exporter can only connect to one replica - not the entire set.
-            password=self.get_secret("app", "monitor_password"),
-            hosts={self.get_hostname_by_unit(self.unit.name)},
-            roles={"monitor"},
-            tls_external=self.tls.get_tls_files("unit") is not None,
-            tls_internal=self.tls.get_tls_files("app") is not None,
-        )
+        hosts = set([self.get_hostname_by_unit(self.unit.name)])
```
Comment (Mehdi-Bendriss) [inline src/charm.py:111]: Same comment about casting to `Set`, considering changing the method to accept a list.

**Review by Mehdi-Bendriss — APPROVED** [review]

File: src/charm.py (hunk 4 above)
Comment (dmitry-ratushnyy) [reply src/charm.py:111]: make sense, updated

File: src/charm.py [hunk 5]
```diff
@@ -375,20 +367,18 @@ def _on_set_password(self, event: ActionEvent) -> None:
...
-            event.fail(
-                f"The action can be run only for users used by the charm: {CHARM_USERS} not {username}."
-            )
+        username = self._get_user_or_fail_event(
+            event, default_username=OperatorUser.get_username()
+        )
+        if not username:
             return
 
         new_password = generate_password()
```
Comment (delgod) [inline src/charm.py:376]: Looks like we need to remove this line

Comment (dmitry-ratushnyy) [reply src/charm.py:376]: done, thank you

**Pull request merged**